Game/MyProject/Game/SceneManagement/Mesh.hpp

```diff
@@ -0,0 +1,56 @@
+//
+//  Mesh.hpp
+//  Game
+//
+//  Created by Tamas Both - (p) on 03/04/2019.
+//
+//  Mesh class stores and manages the vertex and index buffers.
+//  It needs std::vector input for both buffers.
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include <IvVertexBuffer.h>
+#include <IvIndexBuffer.h>
+#include <IvStackAllocator.h>
+#include <IvRenderer.h>
+#include <IvResourceManager.h>
+
+class Mesh : std::enable_shared_from_this<Mesh> {
+private:
+    // private variable(s)
+    IvVertexBuffer* _vertex_buffer;
+    IvIndexBuffer*  _index_buffer;
+    unsigned int    _buffer_size;
+    
+    // private function(s) and method(s)
+    void destroyVertexBuffer();
+    void destroyIndexBuffer();
+    
+public:
+    // constructor(s) and destructor
+    Mesh();
+    ~Mesh();
+    
+    // public functions and methods
+    template <class T>
+    void setVertexBuffer(const std::vector<T>& vertices, const IvVertexFormat& vertex_format) {
+        size_t  current_offset;
+        
+        destroyVertexBuffer();
+        current_offset = IvStackAllocator::mScratchAllocator->GetCurrentOffset();
+        _buffer_size = (unsigned int)vertices.size();
+        
+        _vertex_buffer = IvRenderer::mRenderer->GetResourceManager()->CreateVertexBuffer(vertex_format, (unsigned int)vertices.size(), (void*)vertices.data(), kImmutableUsage);
+        
+        IvStackAllocator::mScratchAllocator->Reset(current_offset);
+    }
+    
+    void setIndexBuffer(const std::vector<unsigned int>&);
+    
+    IvVertexBuffer* getVertexBuffer();
```

Comment:
I'm not sure these should be raw pointers, from what I see this Mesh should **own** the vertex and the index buffer. If someone request raw access they can modify (read delete) the memory. If you want to actually give another instance owner rights, I'd suggest something more descriptive.

Reply:
Mesh should support "updates" this includes getting the vertex buffer and updating the vertices